### ISSUE TITLE
Remove stale instances when toggling 'v-if' with `v-transition`

### ISSUE THREAD

**Issue by butsalt** [vuejs/vue#1045]
[Try this fiddle](http://jsfiddle.net/5sH6A/123/)
Click button, the blue rect should leave in 5s. During this time, click button again to cancel leave transition. Strangely, leave transition doesn't stop. And a new blue rect is appended to the document. This bug might be associated with #1044.

**Comment by yyx990803** [comment]
This is somewhat expected, because `v-if` actually creates a new copy of the content every time it is toggled. But from a UX perspective, it's probably better to instantly remove any stale instances when the condition is changed again.

**Comment by butsalt** [comment]
Gotta, thanks

**Comment by luckymore** [comment]
### scenario：
```html
<!--v-if only work well when the imgUrl is '' -->
<img v-if="imgUrl" v-attr="src: imgUrl" > 
<img v-else src="defaultImg">
```
```javascript
```

**Comment by Thyiad** [comment]
@luckymore 你问的跟这个issues没有关系啊，你这么写不就OK了：v-bind:src="imgUrl || defaultImg" 

？？

**Comment by yyx990803** [comment]
This has been implemented in 2.0. Although we'd like to backport it to 1.x, this is a non-critical change that requires non-trivial effort. Given the bandwidth we have, we are reducing the scope of 1.1 to a number of critical features and low hanging fruits, so unfortunately this will not be implemented.
